lrpar/src/lib/parser.rs

```diff
@@ -231,35 +252,30 @@ where
                 Action::Reduce(pidx) => {
                     let ridx = self.grm.prod_to_rule(pidx);
                     let pop_idx = pstack.len() - self.grm.prod(pidx).len();
-                    let nodes = tstack.drain(pop_idx - 1..).collect::<Vec<Node<StorageT>>>();
-                    tstack.push(Node::Nonterm { ridx, nodes });
 
                     pstack.drain(pop_idx..);
                     let prior = *pstack.last().unwrap();
                     pstack.push(self.stable.goto(prior, ridx).unwrap());
 
                     // Process actions
-                    if let Some((actions, ref mut astack, input)) = actiondata {
-                        action_vec.clear();
-                        action_vec.extend(astack.drain(pop_idx - 1..));
-                        if let Some(f) = actions[usize::from(pidx)] {
-                            astack.push(AStackType::ActionType(f(input, &action_vec)));
-                        }
+                    if let Some(f) = self.actions[usize::from(pidx)] {
+                        let v = AStackType::ActionType(f(ridx, input, astack.drain(pop_idx - 1..)));
+                        astack.push(v);
```

Comment:
Should we squash these two lines into one?

Reply:
Normally I'd agree, but the next commit in the PR reorders this, so it's kind of moot in this case.

Reply:
Fair.